clients/spark-extensions/src/test/java/org/projectnessie/spark/extensions/ITNessieStatements.java

```diff
@@ -0,0 +1,361 @@
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.client.tests.AbstractSparkTest;
+import org.projectnessie.error.BaseNessieClientServerException;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.ImmutableCommitMeta;
+import org.projectnessie.model.ImmutableOperations;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Operations;
+import org.projectnessie.model.Tag;
+
+public class ITNessieStatements extends AbstractSparkTest {
+
+  private String hash;
+  private final String refName = "testBranch";
+
+  @BeforeAll
+  protected static void createDelta() {
+    conf.set(
+        "spark.sql.extensions", "org.projectnessie.spark.extensions.NessieSparkSessionExtensions");
+  }
+
+  @BeforeEach
+  void getHash() throws NessieNotFoundException {
+    hash = nessieClient.getTreeApi().getDefaultBranch().getHash();
+  }
+
+  @AfterEach
+  void removeBranches() throws NessieConflictException, NessieNotFoundException {
+    for (String s : Arrays.asList(refName, "main")) {
+      try {
+        nessieClient
+            .getTreeApi()
+            .deleteBranch(s, nessieClient.getTreeApi().getReferenceByName(s).getHash());
+      } catch (BaseNessieClientServerException e) {
+        // pass
+      }
+    }
+    nessieClient.getTreeApi().createReference(Branch.of("main", null));
+  }
+
+  @Test
+  void testCreateBranchIn() throws NessieNotFoundException {
+
+    List<Object[]> result = sql("CREATE BRANCH %s IN nessie", refName);
+    assertEquals("created branch", row("Branch", refName, hash), result);
+    assertThat(nessieClient.getTreeApi().getReferenceByName(refName))
+        .isEqualTo(Branch.of(refName, hash));
+    result = sql("DROP BRANCH %s IN nessie", refName);
+    assertEquals("deleted branch", row("OK"), result);
+  }
+
+  @Test
+  void testCreateTagIn() throws NessieNotFoundException {
+    List<Object[]> result = sql("CREATE TAG %s IN nessie", refName);
+    assertEquals("created tag", row("Tag", refName, hash), result);
+    assertThat(nessieClient.getTreeApi().getReferenceByName(refName))
+        .isEqualTo(Tag.of(refName, hash));
+    result = sql("DROP TAG %s IN nessie", refName);
+    assertEquals("deleted tag", row("OK"), result);
+    assertThatThrownBy(() -> nessieClient.getTreeApi().getReferenceByName(refName))
+        .isInstanceOf(NessieNotFoundException.class);
```

Comment:
I think it's always a good practice to check for a specific error message as well just to make sure we're getting the right exception message

Reply:
same for all the other places further below

Reply:
fixed